### PR TITLE
first 1 page all translation

### PR DESCRIPTION
**Pull request by koyakei**
part of code comment is not translated yet

### CODE REVIEW

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 1]
```diff
@@ -1,13 +1,393 @@
...
+ブックマークチュートリアル
 ###################
+このチュートリアルは簡単なブックマークのためのアプリを作ります。
+始めるためにはCakeのインストールが必要で、加えて早く作るために
+データベースの作成とCakeが提供するツールを使います。
+
+必要なもの
+
+#. データベースサーバー　この例ではMySqlを使います。SQLについてCreate命令が使えるぐらいの知識.
+　また、``pdo_mysql``が有効になっているか確認して下さい。PDOを使ってCakeはデータベースを動かします。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \`\`pdo_mysql\`\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 2]
```diff
@@ -1,13 +1,393 @@
...
+最も簡単なインストール方法はComposerを使う方法です.  
+Composerを使うとコマンドラインプロンプトやターミナルを使って
+簡単にインストールできます。
+最初にコンポーザーをインストールする必要があります。
+
+cURL がインストールされていたら、このコマンドでインストールされます。::
+
+    curl -s https://getcomposer.org/installer | php
+
+もしくは ``composer.phar`` を`Composer website <https://getcomposer.org/download/>`_から落として下さい.
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space \`Composer website \<https://getcomposer.org/download/\> \`_

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 3]
```diff
@@ -1,13 +1,393 @@
...
+
+
+    php composer.phar create-project --prefer-dist cakephp/app bookmarker
+    
+このコマンドでCake のスケルトン（自動生成されるアプリの基礎）を**bookmarker**　ディレクトリに生成します。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space before \*\*bookmaker\*\*

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 4]
```diff
@@ -1,13 +1,393 @@
...
+を使っていて、例えばここにインストールしていたら、
+C:\\wamp\\www\\dev\\cakephp3)::
+
+    composer create-project --prefer-dist cakephp/app bookmarker
+
+このようにコマンドラインに打って下さい。
+Composerを使うメリットは、 自動的に完全なセットアップをしてくれることです。
+正しい場所に正しい許可でファイルを置くとか、 **config/app.php** を作るとかです。
+
+他にも方法がありますが、コンポーザーを使いたくなかったら:doc:`/installation` を見てください.
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space before :doc:\`installation\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 5]
```diff
@@ -1,13 +1,393 @@
...
+        .gitignore
+        .htaccess
+        .travis.yml
+        composer.json
+        index.php
+        phpunit.xml.dist
+        README.md
+
+それぞれのディレクトリの役割を知りたければここを見て下さい。
+ :doc:`/intro/cakephp-folder-structure`
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: remove space at line head.

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 6]
```diff
@@ -1,13 +1,393 @@
...
+    bin/cake server
 
 .. note::
-    The documentation is not currently supported in Japanese language for this
-    page.
 
-    Please feel free to send us a pull request on
-    `Github <https://github.com/cakephp/docs>`_ or use the **Improve This Doc**
-    button to directly propose your changes.
+    Windows　では、このコマンドは ``bin\cake`` (バックスラッシュではない)です。.
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Perhaps, "バックスラッシュです".

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 7]
```diff
@@ -1,13 +1,393 @@
...
+
+ポート8765でビルトインのPHPサーバーは動きます。ポートを開いて下さい
+**http://localhost:8765** でウェブサーバーのwelcom画面が開くか確かめて下さい。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: "welcome"

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 8]
```diff
@@ -1,13 +1,393 @@
...
+以上の全ての操作でCakePHPはデータベースとの接続以外を可能にします。
+しかし、追加のPHPエクステンションやディレクトリのアクセス許可の変更が必要であればこの限りではあります。
+
+データベースの作成
+=====================
+
+次にDBをセットアップします。まだしていないのならば、空のDBをこのアプリのために作ります。
+名前は例えば ``cake_bookmarks``としておきます.
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \`\`cake_bookmarks\`\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 9]
```diff
@@ -1,13 +1,393 @@
...
+
+    CREATE TABLE bookmarks_tags (
+        bookmark_id INT NOT NULL,
+        tag_id INT NOT NULL,
+        PRIMARY KEY (bookmark_id, tag_id),
+        FOREIGN KEY tag_key(tag_id) REFERENCES tags(id),
+        FOREIGN KEY bookmark_key(bookmark_id) REFERENCES bookmarks(id)
+    );
+
+複合主キーを``bookmarks_tags``で使おうとして注意されると思います。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \`\`bookmarks_tags\`\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 10]
```diff
@@ -1,13 +1,393 @@
...
+Cakeは複合主キーをほとんどどこでもサポートしているので、もっと簡単に複数の
+アプリを一つのアプリで提供できます（マルチテナント）。
+
+テーブルとカラム名は適当に決めずに、CakePHPの
+:doc:`命名規則 </intro/conventions>`に従ったほうがいいです。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space after :doc:\`命名規則 \</intro/conventions\>\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 11]
```diff
@@ -1,13 +1,393 @@
...
+Cakeは十分レガシーなDBに対応できるぐらい柔軟ですが、命名規則に従うことで、時間を節約できます。
+
+
+Database 設定
+======================
+
+次にDBがどこにありどうやって接続するのかCakeに教えます。
+殆どの場合これっきり設定はいじりません。
+
+この設定はとてもわかりやすいはずです。``Datasources.default``のなかにある 配列を in the **config/app.php** にコピペして、
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \`\`Datasources.default\`\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 12]
```diff
@@ -1,13 +1,393 @@
...
+コマンドは以下です:
+
+    // On Windows you'll need to use bin\cake instead.
+    bin/cake bake all users
+    bin/cake bake all bookmarks
+    bin/cake bake all tags
+
+これは、コントローラー、モデル、ビュー、それにタイオするテストケース、と
+ユーザーに対してのフィクスチャー、ブックマークとタグのリソースを一気に生成します。
+サーバを止めてしまった場合、再起動して **http://localhost:8765/bookmarks**に
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space after \*\*http://localhost:8765/bookmarks\*\*

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 13]
```diff
@@ -1,13 +1,393 @@
...
+
+ユーザーを作ると平文でパスワードが保存されてしまいます。セキュリティー上とても良くないので直しましょう
+
+これはまた、モデルレイヤーについて紹介する良い機会です。オブジェクトを操作するメソッドと、
+違うクラスの単一のオブジェクトを分けています。
+一つのレコードに従っている機能が一つの　「エンティティ」クラスにある場合、メソッドはテーブルクラスにあるエンティティーコレクションを操作します。
+
+例えば、パスワードハッシュは個別のレコードで行われ、エンティティーオブジェクトでビヘイビアを操作します。
+なぜなら、毎回パスワードを暗号化するときにmutator/setter　メソッドを使うからです。
+ Cake　は規約ベースのセッターメソッドをエンティーにセットします。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: remove space at line head.

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 14]
```diff
@@ -1,13 +1,393 @@
...
+**src/Model/Entity/User.php**にパスワード用のセッターを追加しましょう。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space after \*\*src/Model/Entity/User.php\*\*

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 15]
```diff
@@ -1,13 +1,393 @@
...
+        {
+            $hasher = new DefaultPasswordHasher();
+            return $hasher->hash($value);
+        }
+    }
+    
+    
+既に作ったユーザーのパスワードを変えたら、前のパスワードに変えて、暗号化されたパスワードが
+表示される。CakePHP `bcrypt
+<http://codahale.com/how-to-safely-store-a-password/>`　でハッシュ化するのがデフォルト設定です。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Replace Fullwidth space to "_" after \`bcrypt \<http://codahale.com/how-to-safely-store-a-password/\>\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 16]
```diff
@@ -1,13 +1,393 @@
...
+
+    Router::scope('/', function ($routes) {
+        // デフォルトのルートに接続.
+        $routes->fallbacks('InflectedRoute');
+    });
+
+上記は新しい　**/bookmarks/tagged/***で``BookmarksController::tags()``に接続するためのルート
+定義したルートでURLをメソッド名とは別に設定できます。
+
+もし**http://localhost:8765/bookmarks/tagged**にアクセスしたらCekeによる有益なエラーメッセージ
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \*\*http://localhost:8765/bookmarks/tagged\*\*

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 17]
```diff
@@ -1,13 +1,393 @@
...
+(the controller action does not exist == アクションの不存在）のある
+エラーページが表示されます。そう表示されたら、 **src/Controller/BookmarksController.php**で
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \*\*src/Controller/BookmarksController.php\*\*

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 18]
```diff
@@ -1,13 +1,393 @@
...
+
+        // Pass variables into the view template context.
+        $this->set([
+            'bookmarks' => $bookmarks,
+            'tags' => $tags
+        ]);
+    }
+
+
+リクエストデータの他の部分にアクセスするためには:`cake-request`
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Perhaps, ":ref:\`cake-request\`"

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 19]
```diff
@@ -1,13 +1,393 @@
...
+を見てください。
+
+Find メソッドの作成
+--------------------------
+
+
+CakePHP ではコントローラーをスリムに保つために、モデルにアプリケーションのろ実行を起きます。
+**/bookmarks/tagged** にアクセスすると、``findTagged()``不存在のエラーが表示されます。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: need space around \`\`findTagged()\`\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 20]
```diff
@@ -1,13 +1,393 @@
...
+**src/Model/Table/BookmarksTable.php** に以下のコードを追加します。:
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: end of line is "::"

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 21]
```diff
@@ -1,13 +1,393 @@
...
+Cakeの finderでは、マッチングするタグを持つブックマークを特定するために``distinct()`` と
+``matching()`` メソッド  を使います。``matching()`` メソッドは、`anonymous function
+<http://php.net/manual/en/functions.anonymous.php>`_を受け付けます。 これは、クエリ日ヅダーの引数を受け付けます。
+コールバック内でクエリビルダーを特定のタグを持つブックマークをフィルターするための
+条件を定義するために使います。
+
+ビューの作成
+-----------------
+
+ **/bookmarks/tagged** にアクセスすると、 ビューファイルがないことを知らせるエラーになります。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Remove space at line head.

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 22]
```diff
@@ -1,13 +1,393 @@
...
+ 次に、ビューファイルを ``tags()`` アクションのために作ります。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Remove space at line head.

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 23]
```diff
@@ -1,13 +1,393 @@
...
+
+            <!-- Use the TextHelper to format text -->
+            <?= $this->Text->autoParagraph($bookmark->description) ?>
+        </article>
+    <?php endforeach; ?>
+    </section>
+
+上記のコードは :doc:`/views/helpers/html` と
+:doc:`/views/helpers/text` を使いました。ヘルパーはビューの生成を助けます。
+また、 HTMLで出力するためのショートカット:php:func:`h` を使いました。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Insert space before :php:func:\`h\`

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 24]
```diff
@@ -1,13 +1,393 @@
...
+ ユーザーデータをに出力するときにHTMLインジェクションを防ぐために
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Remove space at line head.

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 25]
```diff
@@ -1,13 +1,393 @@
...
+ ``h()``を使うことを覚えておいて下さい。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Remove space at line head and insert space after \`\`h()\`\`

Comment (koyakei) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: Thanks reviewing 
Done review all?

File: ja/tutorials-and-examples/bookmarks/intro.rst [hunk 26]
```diff
@@ -1,13 +1,393 @@
...
+
+``tags.ctp`` CakePHPの規約に従ってビューテンプレートファイルから作られます。
```
Comment (suzuki) [inline ja/tutorials-and-examples/bookmarks/intro.rst]: \`\`tags.ctp\`\` **は** CakePHP

**Comment by suzuki** [comment]
@koyakei Thank you your PR. 
I found some of Sphinx syntax miss. Please fix it !

**Comment by suzuki** [comment]
This PR change to #2990 .

**Pull request closed without merging**